lib/logstash/inputs/stdin.rb

```diff
@@ -12,14 +13,18 @@ class LogStash::Inputs::Stdin < LogStash::Inputs::Base
 
   default :codec, "line"
 
-  public
+  def initialize(*args)
+    super(*args)
+    @stop_requested = Concurrent::AtomicBoolean.new(false)
+  end
```

Comment:
no, initializing instance variable in `initialize` is a lot safer, especially if there is no garentees that the `register` method is called and when. I prefer to leave it there.